app/src/main/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivityUnregister.java

```diff
@@ -0,0 +1,145 @@
+    ImageView activityImage;
+    String imagePath;
+    ProgressBar progressBar;
+    Uri uri;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_description);
+
+        Intent intent = getIntent();
+
+        if (intent != null) {
+            act = (Activity) intent.getSerializableExtra("activity");
+        }
+
+        uri = intent.getData();
+        if(uri != null){
+            loadActivityHeaderPicture();
+        }
+
+        createTitleView();
+        createParticipantNumberView();
+        createDescriptionView();
+        createAddressView();
+        createSportView();
+        createDurationView();
+    }
+
+    public void goToLogin(View v) {
+        if (fAuth.getCurrentUser() == null) {
+            startActivity(new Intent(getApplicationContext(), LoginActivity.class)); // sending the user to the "Login" activity
+            finish();
+        }
+    }
+
+    private void createTitleView() {
+        // activityTitle from the activity
+        TextView activityTitle = (TextView) findViewById(R.id.activity_title_description);
+        if (act != null) activityTitle.setText(act.getTitle());
+    }
+
+    private void createParticipantNumberView() {
+        // number of participants from the activity
+        TextView numberParticipantsView = (TextView) findViewById(R.id.activity_number_description);
+        if (act != null) {
+            numberParticipantsView.setText(act.getParticipantId().size() + "/" + act.getNumberParticipant());
+        }
+    }
+
+    private void createDescriptionView() {
+        // description from the activity
+        TextView descriptionView = (TextView) findViewById(R.id.activity_description_description);
+        if (act != null) descriptionView.setText(act.getDescription());
+    }
+
+
+    private void createSportView() {
+        TextView sportView = (TextView) findViewById(R.id.activity_sport_description);
+        if (act != null) {
+            sportView.setText(act.getSport().toString());
+        }
+    }
+
+    private void createDurationView() {
+        TextView durationView = (TextView) findViewById(R.id.activity_duration_description);
+        if (act != null) {
+            durationView.setText(String.valueOf((int) act.getDuration()));
+        }
+    }
+
+    private void createAddressView() {
+        // address from the activity
+        TextView addressView = (TextView) findViewById(R.id.activity_address_description);
+        if (act != null) {
+            addressView.setText(act.getAddress());
+        }
+    }
+
+    private void loadActivityHeaderPicture() {
+        activityImage = findViewById(R.id.activity_image_description);
+        progressBar = findViewById(R.id.progress_bar_activity_description);
+        progressBar.setVisibility(View.VISIBLE);
+        storageReference = FirebaseStorage.getInstance().getReference();
```

Comment:
It would be better to use the providers that Kepler made, no?

Reply:
Probably, once this provider will be ready. Actually I'm not sure how to use the Firebase Storage abstraction here. I know Victor is working on an interface for Firebase Storage in his branch "feature/175_cache_activity_pictures", but you're right, we would definitely have to see how we can implement Firebase Storage abstraction here! This will also be the case wherever we deal with images stored in Firebase Storage (ChatActivity, ProfileActivity, EditProfileActivity, ActivityDescriptionActivity).